File: adapters/go/testdata/pkg/service.go
package pkg

import "example.com/testproject/internal/db"

// Service provides business logic.
type Service struct {
	store *db.Store
}

// NewService creates a new Service.
func NewService() *Service {
	return &Service{store: db.NewStore()}
}

// Hello returns a greeting.
func (s *Service) Hello() string {
	return "hello from service"
}

File: adapters/typescript/testdata/src/service.ts
import { type Storer, newStore } from "./db";

export class Service {
  private store: Storer;

  constructor() {
    this.store = newStore();
  }

  hello(name: string): string {
    const greeting = `Hello, ${name}!`;
    this.store.set(name, greeting);
    return greeting;
  }
}

export function newService(): Service {
  return new Service();
}
